Decoupling: Move WordPress media integration to `wp-story-editor`

mohdsayed opened the pull request:
## Context

Move `wp.media` related code to wp-story-editor

## Summary

<!-- A brief description of what this PR does. -->

## Relevant Technical Choices

<!-- Please describe your changes. -->

## To-do

- [x] Create `MediaUpload` component.
- [x] Move `useMediaPicker` to `wp-story-editor`.
- [x] Use `MediaUpload` in core editor.
- [x] Update all jest tests according to the new code change. ( Provide `MediaUpload` via config )
- [x] Move `getResourceFromAttachment` to `wp-story-editor`.
- [x] Move `getResourceFromMediaPicker` to `wp-story-editor`
- [x] Update relevant object structures after moving `getResourceFromMediaPicker` .
- [x] Update integration test.

## User-facing changes

<!--
Please describe your changes.
Include before/after screenshots or a short video.
-->

## Testing Instructions

<!--
How can the changes in this PR be verified?
Please provide step-by-step instructions how to reproduce the issue, if applicable.
Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->

<!-- ignore-task-list-start -->
- [x] This is a non-user-facing change and requires no QA
<!-- ignore-task-list-end -->

This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

<!-- Examples: new APIs, changes to KSES, etc.  -->

### Does this PR change what data or activity we track or use?

<!-- Examples: changes to telemetry, new third-party APIs -->

### Does this PR have a legal-related impact?

<!-- Examples: new images with unknown sources, new production dependencies with incompatible licenses -->

## Checklist

<!-- Check these after PR creation -->

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

<!--
Please reference the issue(s) this PR addresses.
No URLs, just the issue numbers.
Use "Fixes #123" if it fixes an issue.

NOTE: One reference per line!

Example:

Fixes #123
See #789
-->

Partially addresses https://github.com/google/web-stories-wp/issues/2918
See: https://github.com/google/web-stories-wp/issues/8953

mohdsayed committed:
Move mediaPicker component to wp-story-editor and introduce a new MediaUpload component

mohdsayed committed:
Merge branch 'main' of github.com:google/web-stories-wp into decoupling/wp-media

mohdsayed committed:
Update tests according to code changes

mohdsayed committed:
Merge branch 'main' of github.com:google/web-stories-wp into decoupling/wp-media

mohdsayed committed:
Create MediaUpload dummy component for test

mohdsayed committed:
Update test case for VideoAccessibility

mohdsayed committed:
Use useCallback for MediaUpload render function

mohdsayed committed:
Fix PublishPanel test

mohdsayed committed:
Fix unit test of Panels/Link

mohdsayed committed:
Fix unit tests and update fixture

mohdsayed committed:
Merge branch 'main' of github.com:google/web-stories-wp into decoupling/wp-media

mohdsayed committed:
Update MediaUpload component

mohdsayed committed:
Move getResourceFromAttachment to wp-story-editor

mohdsayed committed:
Update GET_MEDIA_RESPONSE_BODY test data

mohdsayed committed:
Move getResourceFromMediaPicker to wp-story-editor

mohdsayed committed:
Fix indentation

mohdsayed committed:
Update resource keys in onSelect callback

mohdsayed committed:
Update attachment keys in handleChangeTrack

mohdsayed committed:
Update attachment keys after moving getResourceFromMediaPicker

mohdsayed committed:
Merge main and resolve conflicts

mohdsayed committed:
Fix eslint issue

mohdsayed committed:
Merge branch 'main' of github.com:google/web-stories-wp into decoupling/wp-media

mohdsayed committed:
Update test case and fixture according to the new image object structure change

mohdsayed committed:
Update media response with latest changes

mohdsayed committed:
Merge main and resolve conflicts

spacedmonkey commented on packages/story-editor/src/components/panels/design/captions/captions.js:
```diff
@@ -127,11 +126,11 @@ function CaptionsPanel({ selectedElements, pushUpdate }) {
-        trackName: attachment?.filename,
+        track: src,
+        trackId: id,
+        trackName: src.split('/').pop(),
```
Could use use [getFileNameFromUrl](https://github.com/google/web-stories-wp/blob/main/packages/media/src/getFileNameFromUrl.js) here?

mohdsayed replied:
The existing code `attachment?.filename` gives the file name with extension ( for example caption.vtt ) however, [getFileNameFromUrl](https://github.com/google/web-stories-wp/blob/main/packages/media/src/getFileNameFromUrl.js) will return only the file name without extension.

swissspidy replied:
Don‘t think we need to store trackName here in the first place.

mohdsayed committed:
Merge main and resolve conflicts

mohdsayed committed:
Update karma test according to new code changes

mohdsayed committed:
Merge main and resolve conflicts

mohdsayed committed:
Merge main and resolve conflicts

swissspidy committed:
Merge branch 'main' into decoupling/wp-media

swissspidy committed:
Media: Don't force "Upload files" as default tab

This prevents a race condition where the tab is switched after uploading, causing a force-refresh of the items in the "Media Library", causing the just uploaded item to be dropped from the list again.

Updates e2e tests accordingly to first switch to the "Upload files" tab before uploading files.

swissspidy committed:
Use `creationDate` instead of incorrect `uploadDate`

swissspidy committed:
Fix background audio selection callback

swissspidy committed:
Update video poster generation to take resource object
